### ISSUE TITLE
Blob ETAG value is different depending on the API

### ISSUE THREAD

**Issue by SukruthKS** [Azure/azure-storage-java#410]
### Which service(blob, file, queue, table) does this issue concern?
Blob

### Which version of the SDK was used?
10.2.0

### What problem was encountered?
ETAG of a blob is enclosed with quotes when retrieved using `BlobURL.getProperties()` method and without quotes if retrieved using `ContainerURL.listBlobsFlatSegment()` method. I made these calls using a REST client and it appears to be a bug with the API itself, but the previous version of the SDK (8.0.0) masked this by always enclosing ETAG with quotes. Please see the below sample code:

```
String account = "****";
String accessKey = "****";
String containerName = "1cont";
String blobName = "dependency-reduced-pom.xml";

SharedKeyCredentials creds = new SharedKeyCredentials(account, accessKey);
HttpPipeline pipeline = StorageURL.createPipeline(creds);

URL accountURL = new URL("https://" + account + ".blob.core.windows.net");
ServiceURL serviceURL = new ServiceURL(accountURL, pipeline);

ContainerURL containerURL = serviceURL.createContainerURL(containerName);
BlobURL blobURL = containerURL.createBlobURL(blobName);

String eTag = blobURL.getProperties().blockingGet().headers().eTag();
System.out.println("ETAG from getProperties(): " + eTag);

BlobListingDetails listDetails = new BlobListingDetails().withMetadata(true);
ListBlobsOptions options = new ListBlobsOptions().withDetails(listDetails);
List<BlobItem> blobsList = containerURL.listBlobsFlatSegment(null, options)
    .blockingGet().body().segment().blobItems();

for(BlobItem blob : blobsList)
{
    if (!blob.name().equals(blobName))
        continue;

    System.out.println("ETAG from ListBlobs(): " + blob.properties().etag());
    break;
}
```

Output:
```
ETAG from getProperties(): "0x8D64F1915D38ACA"
ETAG from ListBlobs(): 0x8D64F1915D38ACA
```

### Have you found a mitigation/solution?
In my application, I can either always strip or include quotes to get a consistent ETAG value. But it would be good if the SDK handled this so not every application has to do this. The best solution would be to fix the API itself.

**Comment by rickle-msft** [comment]
Hi, @SukruthKS. Thank you for catching this inconsistency and reporting it. I've verified it on my end, too. We should be able to do something to make this less of a hassle to work with.

**Comment by rickle-msft** [comment]
This should be resolved in v11.0.0 which is now out. I will close this issue as I believe it has been addressed, but please feel free to reopen it if you continue to experience issues.
